lib/errors/oauth-error.js

```diff
@@ -9,33 +9,33 @@ const http = require('http');
  * Constructor.
  */
 
-function OAuthError(messageOrError, properties) {
-  let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
-  const error = messageOrError instanceof Error ? messageOrError : null;
-  if (properties == null || !Object.entries(properties).length ) {
-    properties = {};
-  }
+class OAuthError extends Error {
+	constructor(messageOrError, properties) {
+		let message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
+		const error = messageOrError instanceof Error ? messageOrError : null;
+		if (properties == null || !Object.entries(properties).length) {
+			properties = {};
+		}
 
-  properties = Object.assign({ code: 500 }, properties);
+		properties = Object.assign({ code: 500 }, properties);
 
-  if (error) {
-    properties.inner = error;
-  }
-  if (!message || message.length === 0) {
-    message = http.STATUS_CODES[properties.code];
-  }
-  this.code = this.status = this.statusCode = properties.code;
-  this.message = message;
-  for (const key in properties) {
-    if (key !== 'code') {
-      this[key] = properties[key];
-    }
-  }
-  Error.captureStackTrace(this, OAuthError);
+		if (error) {
+			properties.inner = error;
+		}
+		if (!message || message.length === 0) {
+			message = http.STATUS_CODES[properties.code];
+		}
+		this.code = this.status = this.statusCode = properties.code;
+		this.message = message;
+		for (const key in properties) {
+			if (key !== 'code') {
+				this[key] = properties[key];
+			}
+		}
+		Error.captureStackTrace(this, OAuthError);
```

Comment:
is this actually necessaray? I think there is actually an issue regarding this unnecessary captureStackTrace call in original repo. As we have now a proper class, the stack trace should be already fine.

Reply:
I just found this issue

https://github.com/oauthjs/node-oauth2-server/issues/436

Reply:
I suppose that when extending the Error class the act of capturing the stack trace is already done by the Error class constructor.

I will remove it, but I'm not entirely sure about it

Reply:
I think we should add at least one unit test to check, whether the correct stacktrace is added when raising errors. It could be that `super(message, properties);` messes up the prototype chain so please add one test.